themes/default/content/blog/advanced-aws-networking-part-2/index.md

```diff
@@ -0,0 +1,534 @@
+
+# The meta_image appears in social-media previews and on the blog home page.
+# A placeholder image representing the recommended format, dimensions and aspect ratio
+# has been provided for you.
+meta_image: meta.png
+
+# At least one author is required.
+# The values in this list correspond with the `id` properties
+# of the team member files at /data/team/team.
+# Create a file for yourself if you don't already have one.
+authors:
+    - josh-kodroff
+    - andy-taylor
+    - jose-juhala
+
+# At least one tag is required.
+# Lowercase, hyphen-delimited is recommended.
+tags:
+    - aws
+    - networking
+    - hub-and-spoke
+    - python
+
+# See the blogging docs at https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md.
+# for additional details,
+# and please remove these comments before submitting for review.
+---
+
+In this blog series you will learn how to create a hub-and-spoke network architecture in AWS with centralized egress and traffic inspection. In this second installment, we'll show you how to create spoke VPCs to run your workloads, verify centralized egress is working, and then add centralized traffic inspection using Pulumi, the infrastructure as code tool that enables you to manage infrastructure with real programming languages!
+
+<!--more-->
+
+## Introduction
+
+In [AWS Advanced Networking Part 1](https://pulumi.com/blog/advanced-aws-networking-part-1/):
+
+1. Discussed the benefits of a hub-and-spoke network architecture with centralized egress.
+1. Created the shared [AWS Transit Gateway](https://aws.amazon.com/transit-gateway/) resources to be shared among our spoke VPCs and inspection VPC.
+1. Defined our inspection VPC as a Pulumi [component resource](https://www.pulumi.com/docs/intro/concepts/resources/components/).
+
+Now, to finish the architecture, we will need to:
+
+1. Create a spoke VPC as a Pulumi component resource.
+1. Test the network architecture by verifying that centralized egress is working.
+1. Add traffic inspection with [AWS Network Firewall](https://aws.amazon.com/network-firewall/).
+
+{{% notes type="info" %}}
+In subsequent sections we will be showing code samples. Because this is a more advanced Pulumi program, we will be focusing on the most important parts of the architecture rather than covering every single line in the codebase. The complete code for this project is available on [GitHub](https://github.com/pulumi/examples/tree/master/aws-py-hub-and-spoke-network), which is a great place to go for full context.
+{{% /notes %}}
+
+## Spoke VPC
+
+Like your inspection VPC, we will implement the spoke VPC as a Pulumi component resource. Component resources allow you to logically group resources to form reusable components. Because your network will contain multiple spoke VPCs (at least 2, possibly more), and the spoke VPC will contain both the basic VPC as well as connectivity to Transit Gateway, the spoke VPC is a natural fit for a component resource.
```

Comment:
```suggestion
Like your inspection VPC, we will implement the spoke VPC as a Pulumi component resource. Component resources allow you to logically group resources to form reusable components. Because your network will contain multiple spoke VPCs (at least 2, possibly more), and the spoke VPC will contain both the basic VPC and connectivity to Transit Gateway, the spoke VPC is a natural fit for a component resource.
```